nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -0,0 +1,188 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.platform.comment.api.Comment;
+import org.nuxeo.ecm.platform.comment.api.CommentImpl;
+import org.nuxeo.ecm.platform.comment.api.CommentManager;
+import org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants;
+import org.nuxeo.ecm.platform.query.api.PageProvider;
+import org.nuxeo.ecm.platform.query.api.PageProviderService;
+import org.nuxeo.ecm.platform.query.nxql.CoreQueryAndFetchPageProvider;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Comment service implementation. The comments are linked together thanks to a parent document id property.
+ * 
+ * @since 10.3
+ */
+public class PropertyCommentManager implements CommentManager {
+
+    private static final Log log = LogFactory.getLog(PropertyCommentManager.class);
+
+    protected static final String GET_COMMENTS_FOR_DOC_PAGEPROVIDER_NAME = "GET_COMMENTS_FOR_DOCUMENT";
+
+    @Override
+    public List<DocumentModel> getComments(DocumentModel docModel) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DocumentModel> getComments(DocumentModel docModel, DocumentModel parent) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, String comment) {
+        return createComment(docModel, comment, docModel.getCoreSession().getPrincipal().getName());
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, String text, String author) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, DocumentModel comment) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, DocumentModel parent, DocumentModel child) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void deleteComment(DocumentModel docModel, DocumentModel comment) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<DocumentModel> getDocumentsForComment(DocumentModel comment) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocumentModel getThreadForComment(DocumentModel comment) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DocumentModel createLocatedComment(DocumentModel docModel, DocumentModel comment, String path) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String createComment(CoreSession session, Comment comment) {
+        return CoreInstance.doPrivileged(session, s -> {
```

Comment:
What's the security story for this? It seems anyone will be able to create comments on any document. Where are the security checks going to be done?

Reply:
It is because I kept the same behaviour as what we did for the annotations, minus the folder creation.
I'm thinking of adding a TODO to discuss how we should store the comments with this new implementation (in the scope of https://jira.nuxeo.com/browse/NXP-25425 for ex).Maybe we should keep the same logic as what was done with the annotations ?